### PR TITLE
[WLI] SQL azure workbooks changes. 

### PR DESCRIPTION
**Pull request by heyaWorld**
## PR Checklist
Using Telegraf config for SQL azure
Changing how tags are stored and other minor fixes.
Add multi database support
Support for additional regions
Other minor fixes for SQL azure and changes in Custom logs.

image

* [x] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
* [x] *if updating templates, it is helpful to post a screenshot of what the changes look like.*
* [x] *if updating templates, ensure that your parameters and steps have meaningful names.*
* [ ] *if adding new templates, or changing items in a gallery, it's helpful to post a screenshot of what the gallery looks like now*

### CODE REVIEW

File: Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook [hunk 1]
```diff
@@ -25,9 +36,12 @@
...
             "name": "SQLServer",
             "label": "SQL Server",
-            "type": 5,
+            "type": 2,
             "isRequired": true,
-            "query": "Resources\n| where id =~ '{SqlDatabase}'\n| project SqlServer = extract('(.+/microsoft.sql/(managedinstances|servers)/.+)/databases/.+', 1, tolower(id))\n| project value = SqlServer, label = SqlServer, selected = true",
+            "multiSelect": true,
+            "quote": "\"",
+            "delimiter": ",\\r\\n    ",
+            "query": "Resources\n| where id in~ ({SqlDatabase})\n| project SqlServer = extract('.+/microsoft.sql/(managedinstances|servers)/(.+)/databases/.+', 2, tolower(id)), Db = extract('.+/microsoft.sql/(managedinstances|servers)/.+/databases/(.+)', 2, tolower(id))\n| project value = strcat('Server=', SqlServer, '.database.windows.net;Port=1433;database=', Db, ';User Id=<user>;Password=<pw>;app name=telegraf;log=1;'), label = SqlServer, selected = true",
```
Comment (acearun) [inline Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook:44]: Doug found a way to do this without tolower(id). You use the (?i) to do case invariant comparisons in the regex.

File: Workbooks/Workloads/Collection logs/Collection logs.workbook [hunk 2]
```diff
@@ -118,7 +118,7 @@
...
+        "query": "union telegraflogs_CL, waagentlogs_CL\n| order by TimeGenerated desc\n| extend Message = iff(Type == 'telegraflogs_CL', iff('{LogsFilter}' == \"All Logs\", extract(@'.+Z D! \\[outputs.socket_writer\\] (.+)', 1, RawData), \nextract(@'.+Z E! (.+)', 1, RawData)), iff('{LogsFilter}' == \"All Logs\", extract(@'.+Z (.+)', 1, RawData), extract(@'.+Z (.+(Microsoft\\.Azure\\.Monitor\\.WorkloadInsights\\.Test\\.Workload\\.LinuxConfigAgent|Microsoft\\.Azure\\.Monitor\\.WorkloadInsights\\.Test\\.Workload\\.LinuxInstallerAgent)(.+))', 1, RawData) ) )\n| where ('{LogsFilter}' == \"Debug Logs\" and Message != \"\") or ('{LogsFilter}' == \"All Logs\")\n| project Time = TimeGenerated, Message = iff(Message == '', RawData, Message), ['Log type'] = iff(Type == 'telegraflogs_CL', 'Telegraf', iff('{LogsFilter}' == \"Debug Logs\", \"VMExtension\", 'WA Agent'))\n| take 500",
```
Comment (acearun) [inline Workbooks/Workloads/Collection logs/Collection logs.workbook:121]: What is VMExtension in the Log Type column here supposed to mean?

File: Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook [hunk 3]
```diff
@@ -438,6 +296,77 @@
...
+                      "value::all"
+                    ],
+                    "parameters": [
+                      {
+                        "id": "573cff77-860f-4b27-bfa4-1ca675a13b98",
+                        "version": "KqlParameterItem/1.0",
+                        "name": "SqlTelegrafConfig",
+                        "label": "Sql Telegraf config",
+                        "type": 1,
+                        "query": "Resources\r\n| take 1\r\n| project toml = strcat('[[inputs.sqlserver]]\\r\\n  servers = [\\r\\n    {SQLServer}\\r\\n  ]\\r\\n\\r\\n  query_version = 2\\r\\n  azuredb=true\\r\\n  exclude_query = [ \\'Schedulers\\' , \\'SqlRequests\\']\\r\\n\\r\\n  [inputs.sqlserver.tags]\\r\\n    insight = \"sqlaz\"')",
```
Comment (acearun) [inline Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook:323]: I wonder if the insight tags should not be 'sqlaz' be ‘sql-azure’ like in the insight workbook.

File: Workbooks/Workloads/Collection logs/Collection logs.workbook (hunk 2 above)
Comment (heyaWorld) [reply Workbooks/Workloads/Collection logs/Collection logs.workbook:121]: We filter the WA Agent logs to show only our "VM extension" logs when user select the Debug logs option. So by default there are two types of logs: Telegraf and Waagent, and filtered logs are named: Telegraf and VMExtension.

Basically to differentiate between the filtered and all logs.

Comment (acearun) [reply Workbooks/Workloads/Collection logs/Collection logs.workbook:121]: This is confusing. Maybe rename VMExtension -> Telegraf setup.

File: Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook (hunk 3 above)
Comment (heyaWorld) [reply Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook:323]: yes, it should be 'sql-azure', will update.

File: Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook (hunk 1 above)
Comment (heyaWorld) [reply Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook]: Fixed

File: Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook (hunk 3 above)
Comment (heyaWorld) [reply Workbooks/Workloads/Onboard SQL Azure Insights/Onboard SQL Azure Insights.workbook]: fixed

File: Workbooks/Workloads/Collection logs/Collection logs.workbook (hunk 2 above)
Comment (heyaWorld) [reply Workbooks/Workloads/Collection logs/Collection logs.workbook]: fixed

**Pull request merged**